src/openproduct/producten/viewsets/product.py

```diff
@@ -202,27 +209,40 @@ class Meta:
     ),
 )
 class ProductViewSet(AuditTrailViewSetMixin, NotificationViewSetMixin, ModelViewSet):
-    queryset = Product.objects.prefetch_related(
-        "eigenaren",
-        "documenten",
-        "taken",
-        "zaken",
-        Prefetch(
-            "producttype",
-            queryset=ProductType.objects.select_related(
-                "uniforme_product_naam"
-            ).prefetch_related(
-                "translations",
-                Prefetch(
-                    "themas", queryset=Thema.objects.select_related("hoofd_thema")
-                ),
-            ),
-        ),
-    )
+    queryset = Product.objects.all()
     lookup_field = "uuid"
     serializer_class = ProductSerializer
     filterset_class = ProductFilterSet
     notifications_kanaal = KANAAL_PRODUCTEN
+    permission_classes = [ProductTypeObjectPermission, DjangoModelPermissions]
+
+    def get_queryset(self):
+        if self.action != "list" or self.request.user.is_superuser:
```

Comment:
This does mean that you can retrieve products that you do not have permission for, I think